models/Document/Service.php

```diff
@@ -426,7 +426,7 @@ public static function getByUrl(string $url): ?Document
                 $sitesObjects = $sitesList->load();
```

Comment:
it seems that it could be micro-optimized if 
`if (!$document) {` on line `424` is changed to 
`if (!$document && isset($urlParts['host'])) {`
so it would skip the Listing load and consequent `foreach`, rather than checking if isset on every site

wdyt?

Reply:
i've pushed the suggested change on https://github.com/pimcore/pimcore/pull/17259/commits/0947a0b4201b351cb46b4ae84b92df5827459d1b, hope you won't mind it

Does it also look good for you or am i missing anything? Thank you in advance